moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1430,6 +1437,61 @@ bool TrajectoryExecutionManager::executePart(std::size_t part_index)
   }
 }
 
+bool TrajectoryExecutionManager::waitForRobotToStop(const TrajectoryExecutionContext &context, double wait_time)
+{
+  if (allowed_start_tolerance_ == 0)  // skip validation on this magic number
+    return true;
+
+  ros::WallTime start = ros::WallTime::now();
+  double time_remaining = wait_time;
+
+  robot_state::RobotStatePtr prev_state, cur_state;
+  prev_state = csm_->getCurrentState();
+  prev_state->enforceBounds();
+
+  // assume robot stopped when 3 consecutive checks yield the same robot state
+  unsigned int no_motion_count = 0;  // count iterations with no motion
+  while (time_remaining > 0. && no_motion_count < 3)
+  {
+    if (!csm_->waitForCurrentState(ros::Time::now(), time_remaining) || !(cur_state = csm_->getCurrentState()))
+    {
+      ROS_WARN_NAMED("traj_execution", "Failed to receive current joint state");
+      return false;
+    }
+    cur_state->enforceBounds();
+    time_remaining = wait_time - (ros::WallTime::now() - start).toSec();  // remaining wait_time
+
+    // check for motion in effected joints of execution context
+    bool moved = false;
+    for (const auto &trajectory : context.trajectory_parts_)
```

Comment:
do you plan on picking this to previous ROS versions? if so we should avoid ``auto``